src/Modal.js

```diff
@@ -120,6 +120,10 @@ class Modal extends React.Component {
     if (this.props.autoFocus && this.state.isOpen && !prevState.isOpen) {
       this.setFocus();
     }
+
+    if (prevProps.zIndex !== this.props.zIndex) {
```

Comment:
Probably also want to check to see if `this._element` exists. Also, it would be nice to add a test to ensure this functionality isn't lost in future changes.

Reply:
did what you asked